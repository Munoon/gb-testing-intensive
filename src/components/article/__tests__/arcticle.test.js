import React from "react";
import { render, screen } from "@testing-library/react";
import Article from "../index";

describe("article test", () => {
    it('snapshot', () => {
        const article = {
            publishedAt: new Date(''),
            title: 'Test Article',
            imageUrl: 'http://localhost/img',
            url: 'http://localhost/article',
            summary: 'Text about this article',
            newsSite: 'CCB'
        };
        const component = render(<Article article={article} />);
        expect(component).toMatchSnapshot();
    });

    it('should render placeholder', () => {
        render(<Article />);
        const element = screen.findByTestId('article-placeholder')
        expect(element).toBeDefined()
    });
});