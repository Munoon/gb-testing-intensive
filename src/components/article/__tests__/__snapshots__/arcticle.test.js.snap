// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="post"
        href="http://localhost/article"
      >
        <div
          class="header"
        >
          Test Article
        </div>
        <div
          class="imageWrapper"
        >
          <img
            alt="Test Article"
            class="image"
            src="http://localhost/img"
          />
        </div>
        <div
          class="summary"
        >
          Text about this article
        </div>
        <div
          class="bottom"
        >
          <div
            class="source"
          >
            CCB
          </div>
          <div
            class="date"
          >
            Invalid Date
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="post"
      href="http://localhost/article"
    >
      <div
        class="header"
      >
        Test Article
      </div>
      <div
        class="imageWrapper"
      >
        <img
          alt="Test Article"
          class="image"
          src="http://localhost/img"
        />
      </div>
      <div
        class="summary"
      >
        Text about this article
      </div>
      <div
        class="bottom"
      >
        <div
          class="source"
        >
          CCB
        </div>
        <div
          class="date"
        >
          Invalid Date
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
